### PR TITLE
lib: modem_info: Fix coverity issues

### PR DESCRIPTION
**Pull request by rlubos**
Address issues found by Coverity

### CODE REVIEW

File: lib/modem_info/modem_info_json.c [hunk 1]
```diff
@@ -277,6 +286,9 @@ int modem_info_json_string_encode(struct modem_param_info *modem,
 
 		total_len += ret;
 		json_add_obj(root_obj, MODEM_INFO_JSON_KEY_DEV_INF, device_obj);
+
+		/* device_obj is not part of the root_obj */
```
Comment (jayteemo) [inline lib/modem_info/modem_info_json.c:290]:
```suggestion
		/* device_obj is now part of the root_obj */
```

File: lib/modem_info/modem_info_json.c [hunk 2]
```diff
@@ -266,6 +272,9 @@ int modem_info_json_string_encode(struct modem_param_info *modem,
 
 		total_len += ret;
 		json_add_obj(root_obj, MODEM_INFO_JSON_KEY_SIM_INF, sim_obj);
+
+		/* sim_obj is not part of the root_obj */
```
Comment (jayteemo) [inline lib/modem_info/modem_info_json.c:276]:
```suggestion
		/* sim_obj is now part of the root_obj */
```

File: lib/modem_info/modem_info_json.c [hunk 3]
```diff
@@ -255,6 +258,9 @@ int modem_info_json_string_encode(struct modem_param_info *modem,
 
 		total_len += ret;
 		json_add_obj(root_obj, MODEM_INFO_JSON_KEY_NET_INF, network_obj);
+
+		/* network_obj is not part of the root_obj */
```
Comment (jayteemo) [inline lib/modem_info/modem_info_json.c:262]:
```suggestion
		/* network_obj is now part of the root_obj */
```

File: lib/modem_info/modem_info_json.c (hunk 1 above)
Comment (rlubos) [reply lib/modem_info/modem_info_json.c]: Thanks, I've fixed all of the reported typos.

**Review by lemrey — APPROVED** [review]

**Review by jayteemo — APPROVED** [review]

**Pull request merged**